network/p2p/gossip/bloom.go

```diff
@@ -65,11 +65,15 @@ func ResetBloomFilterIfNeeded(
 	bloomFilter *BloomFilter,
 	falsePositiveProbability float64,
 ) (bool, error) {
-	if bloomFilter.bloom.FalsePositiveProbability() < falsePositiveProbability {
+	numSeeds, numBytes := bloomFilter.bloom.Parameters()
+	// TODO: Precalculate maxEntries, as it is independent of the current state
+	// of the bloom filter.
+	maxEntries := bloom.EstimateEntries(numSeeds, numBytes, falsePositiveProbability)
+	if bloomFilter.bloom.Count() < maxEntries {
```

Comment:
If we are perpetually at `maxEntries`, I wonder if we should be rate limiting how much this gets reset?

Reply:
I think this `gossip/bloom` would be significantly more powerful if we were able to completely hide the size of the bloom filter from the consumer of this package.

Essentially I think we should have a dynamically re-sizing bloom filter based on the size of the set we are working over. If we are resetting a lot -> bigger filter. If we aren't resetting much -> smaller filter.

Reply:
Doing here: https://github.com/ava-labs/avalanchego/pull/2591